powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/replay/DefaultReplayVerificationService.java

```diff
@@ -52,9 +52,15 @@ class DefaultReplayVerificationService implements ReplayVerificationService {
     private final PowerAuthServiceConfiguration config;
 
     @Override
-    public void checkAndPersistUniqueValue(UniqueValueType type, Date requestTimestamp, String ephemeralPublicKey, String nonce, String identifier) throws GenericServiceException {
+    public void checkAndPersistUniqueValue(UniqueValueType type, Date requestTimestamp, String ephemeralPublicKey, String nonce, String identifier, String version) throws GenericServiceException {
         logger.debug("Checking and persisting unique value, request type: {}, identifier: {}", type, identifier);
-        final Date expiration = Date.from(Instant.now().plus(config.getRequestExpirationInMilliseconds(), ChronoUnit.MILLIS));
+        final int requestExpiration;
+        if (!"3.2".equals(version)) {
```

Comment:
... this does not fix the issue raised in my original comment, we will still need to adjust the code here in case version changes which is tricky.

Reply:
So, how do you suggest to fix it?

Reply:
I think I see now what you mean.